test/dataset_utils.py

```diff
@@ -2,14 +2,18 @@
 import unittest
 from typing import Any, Dict, Union
 from unittest.mock import patch
-
+import pytest
 import numpy as np
+import os
+
+# Location of the files to be saved and extracted by the datasets during testing
+TEST_LOCATION_ON_SYSTEM = "~/../../tmp"
+TEST_LOCATION_ON_SYSTEM = os.path.expanduser(TEST_LOCATION_ON_SYSTEM)
```

Comment:
such hardcoded paths seem extremely brittle, maybe it would be worth adding a separate test for a path that includes a tilde